### PR TITLE
Cloning has a small chance to give you random mutations

### PR DESCRIPTION
**Pull request by Bod9001**
2.5% for each body part that is damaged 
because 25 damages applied to each body part

### Changelog:
CL: [New] Cloning has a small chance to give you random mutations negative and positive

### CODE REVIEW

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs [hunk 1]
```diff
@@ -183,6 +183,12 @@ public float TotalDamage
 			}
 		}
 
+		/// <summary>
+		/// So functions can subscribe and do there Funky stuff when damage is taken
+		/// </summary>
+		public event Action<AttackType,DamageType, float> OnDamageChangeBy;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]: I can't believe we didn't have this event already. I have some suggestions tho.
```suggestion
		public event Action<AttackType,DamageType, float> OnDamageTaken;
```
Current name suggests you can get who did the damage or so I understand.

File: UnityProject/Assets/Scripts/HealthV2/Living/BodyPartMutations.cs [hunk 2]
```diff
@@ -38,6 +38,43 @@ public static MutationRoundData GetMutationRoundData(MutationSO Mutation)
...
+			//Range = -999999 to 9999999
+			if (amount > 0)
+			{
+				//Range = 0 to 9999999
+				amount = Mathf.Clamp(amount, 0, 100);
+				//Range = 0 to 100
+				//Percentage 100 = 10
+
+				var RNG= Random.Range(0, 1000);
+				if (amount >= RNG)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/BodyPartMutations.cs:61]: amount can only be ``100`` at most. Why make the range from 0 to 1000?

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]: it describes the change in damage?

File: UnityProject/Assets/Scripts/HealthV2/Living/BodyPartMutations.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/BodyPartMutations.cs:61]: so 100 = 10% chance of happening

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]: maybe change isn't the right word, uhmm

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs [hunk 3]
```diff
@@ -183,6 +183,12 @@ public float TotalDamage
 			}
 		}
 
+		/// <summary>
+		/// So functions can subscribe and do there Funky stuff when damage is taken
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]:
```suggestion
		/// Triggers when a body part receives damage. 
		///It has the attack type, damage type and the amount of damage as parameters for the callback
```

**Review by corp-0 — APPROVED** [review]

**Pull request merged**